modules/phase_field/include/materials/EpsAndMAnisoGauss.h

```diff
@@ -0,0 +1,143 @@
+  std::vector<double> _vbalibx, _vbaliby, _vbalibz, _thetabalib, _miubalibx, _miubaliby, _miubalibz,
+      _MinimaEnergy;
+
+  // Names of material properties.
+  const MaterialPropertyName _eps_name;        // The name of the anisotropic epsilon - AD.
+  const MaterialPropertyName _eps_EN_name;     // The name of the anisotropic epsilon - NoAD.
+  const MaterialPropertyName _deps_minus_name; // Negative derivatives of epsilon - AD.
+  const MaterialPropertyName _deps_plus_name;  // Positive derivatives of epsilon - AD.
+  const MaterialPropertyName
+      _VlibR_name; // The name of the library boundary normals in the simulation reference frame.
+  const MaterialPropertyName
+      _Vec_name; // The name of the normalized gradients in the simulation reference frame.
+  const MaterialPropertyName _dm_minus_name;     // Negative derivatives of m - AD.
+  const MaterialPropertyName _dm_plus_name;      // Positive derivatives of m - AD.
+  const MaterialPropertyName _S_switch_name;     // The name of the switch property.
+  const MaterialPropertyName _m_name;            // The name of the anisotropic m - AD.
+  const MaterialPropertyName _m_EN_name;         // The name of the anisotropic m - NoAD.
+  const MaterialPropertyName _L_name;            // The name of L - AD.
+  const MaterialPropertyName _L_EN_name;         // The name of L - NoAD.
+  const MaterialPropertyName _sigma_name;        // Grain boundary energy in eV/nm^2.
+  const MaterialPropertyName _sigmaORIUNIT_name; // Grain boundary energy in J/m^2.
+  const MaterialPropertyName _qwg_name;          // w component of the quaternion.
+  const MaterialPropertyName _qxg_name;          // x component of the quaternion.
+  const MaterialPropertyName _qyg_name;          // y component of the quaternion.
+  const MaterialPropertyName _qzg_name;          // z component of the quaternion.
+
+  // Epsilon.
+  ADMaterialProperty<Real> & _eps;
+  MaterialProperty<Real> & _eps_EN;
+
+  // Derivatives of epsilon.
+  ADMaterialProperty<RealGradient> & _deps_minus;
+  ADMaterialProperty<RealGradient> & _deps_plus;
+
+  // Switch for anisotropy calculation.
+  ADMaterialProperty<Real> & _S_switch;
+
+  // Library boundary normals in the simulation reference frame.
+  ADMaterialProperty<RealGradient> & _VlibR;
+
+  // Normalized gradients in the simulation reference frame.
+  ADMaterialProperty<RealGradient> & _Vec;
+
+  // Derivatives of m.
+  ADMaterialProperty<RealGradient> & _dm_minus;
+  ADMaterialProperty<RealGradient> & _dm_plus;
+
+  // Whether to add Gaussian to epsilon.
+  const bool & _ADDGaussian;
+
+  //  m.
+  ADMaterialProperty<Real> & _m;
+  MaterialProperty<Real> & _m_EN;
+
+  // Mobility property.
+  ADMaterialProperty<Real> & _L;
+  MaterialProperty<Real> & _L_EN;
+
+  // Grain boundary energy properties eV/nm^2 and J//m^2.
+  ADMaterialProperty<Real> & _sigma;
+  ADMaterialProperty<Real> & _sigmaORIUNIT;
+
+  // Quaternion components
+  ADMaterialProperty<Real> & _qwg;
```

Comment:
I ended merging all four materials into a single file

I tried to cut the routine up but since the innermost loop still uses values from the outermost loop, I kept having issues.